### ISSUE TITLE
Initial Setup

### ISSUE THREAD

**Issue by ZimboPro** [jagregory/cognito-local#10]
I am a noob to AWS and hence struggling to set up a CognitoUserPool.

I am using `amazon-cognito-identity-js` to sign up, sign in etc but when I use run `cognito-local` without any changes to the default values I get the `ResourceNotFoundException` exception.

I created a script to use `aws-sdk` and `CognitoIdentityServiceProvider`
``` javascript
const service = new CognitoIdentityServiceProvider({
    credentials: {
      accessKeyId: "local",
      secretAccessKey: "local",
    },
    region: "local",
    endpoint: `http://localhost:9229`,
  })
```
and then create a UserPoolClient
``` javascript
service.createUserPoolClient({
    ClientName: 'local',
    UserPoolId: 'local_host' // amazon-cognito-identity-js fails if it is just 'local'
}, function(err, data) {
    if (err) {
        console.log('Error');
        console.log(err);
    } else {
        console.log('Success');
        console.log(data);
    }
})
``` 
It succeeds in creating a UserPoolClient
clients.json file
```json
{
  "Clients": {},
  "Clients,dl5si5mdk7o8h1zzotc93cl7u": {
    "ClientId": "dl5si5mdk7o8h1zzotc93cl7u",
    "ClientName": "local",
    "UserPoolId": "local_host",
    "CreationDate": 1604425829903,
    "LastModifiedDate": 1604425829903,
    "AllowedOAuthFlowsUserPoolClient": false,
    "RefreshTokenValidity": 30
  }
}
```
but even when I update `amazon-cognito-identity-js` accordingly, I still get the `ResourceNotFoundException` error
```javascript
const poolData: ICognitoUserPoolData = {
  UserPoolId: 'local_host', // amazon-cognito-identity-js fails if it is just 'local'
  ClientId: 'dl5si5mdk7o8h1zzotc93cl7u',
  endpoint = 'http://localhost:9229'
};
```

Error stack in the termial
```
Error handling target: SignUp ResourceNotFoundError: Resource not found
    at Object.getUserPoolForClientId (project/node_modules/cognito-local/lib/services/cognitoClient.js:25:15)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async project/node_modules/cognito-local/lib/targets/signUp.js:23:20
    at async project/node_modules/cognito-local/lib/server/server.js:69:22 {
  code: 'CognitoLocal#ResourceNotFoundException'
}
```

cognito-local: 2.1.0
Node version: 10.3
amazon-cognito-identity-js: 4.5.0
OS: Debian

Any suggestions or tips will be helpful

**Comment by ZimboPro** [comment]
Managed to figure it out. So it seems an error occurred and created the `clients.json` as shown above. After deleting the json files and running the script to create a UserPoolClient it created the file properly and solved the problem

**Comment by mayeryuca** [comment]
@ZimboPro Could you post your corrected `clients.json` file? I'm having the exact same error here and can't solve even after deleting all files and running the script again.

**Comment by ZimboPro** [comment]
@mayeryuca  here is the created JSON file and the full script that created the file
`clients.json`
```json
{
  "Clients": {
    "vla4cwe66kb2k1u0kj37albj": {
      "ClientId": "vla4cwe66kb2k1u0kj37albj",
      "ClientName": "local",
      "UserPoolId": "local_host",
      "CreationDate": 1604598026711,
      "LastModifiedDate": 1604598026711,
      "AllowedOAuthFlowsUserPoolClient": false,
      "RefreshTokenValidity": 30
    }
  }
}
```
Script
```javascript
const localCognito = require('aws-sdk')

const cognito = new localCognito.CognitoIdentityServiceProvider({
    credentials: {
      accessKeyId: "local",
      secretAccessKey: "local",
    },
    region: "local",
    endpoint: `http://localhost:9229`
  });
cognito.createUserPoolClient({
    ClientName: 'local',
    UserPoolId: 'local_host'
}, function(err, data) {
    if (err) {
        console.log('Error');
        console.log(err);
    } else {
        console.log('Success');
        console.log(data);
    }
})
```
For the script just need to make sure that cognito-local is running

**Comment by LordAndreasOtten** [comment]
Hmm i'm having the same issue even with the same code and same node + plugin version

**Comment by jagregory** [comment]
Hopefully the issues you all ran into here have been fixed in [v2.3.0](https://github.com/jagregory/cognito-local/releases/tag/v2.3.0), which included a fix for the client storage getting mangled.
